training-service/src/app/content/subcategory/service/subcategory.service.ts

```diff
@@ -1,19 +1,43 @@
 import { Injectable } from '@angular/core';
 import { Subcategory } from '../model/subcategory';
 import { subcategory } from '../../../sample-data/subcategory';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
+import { throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SubcategoryService {
 
   private subcatList: Subcategory[] = subcategory;
+  private readonly apiUrl: string = environment.apiUrl;
+
+  msgTrue = false;
+
+  private handleError(error: HttpErrorResponse) {
```

Comment:
Please add proper TODO comments here, 
https://medium.com/imdoneio/the-imdone-pitch-feedback-welcome-386430accf01

Reply:
done